### DISCUSSION TITLE
Minestom with unit test

### DISCUSSION THREAD

**Discussion by highright1234** [Minestom/Minestom#776] [Help]
I want to use Minestome with a unit test, but I can't find the documentation. Is it have any related documentation for test?

**Comment by TheMode** [comment]
Our testing API is currently still internal. We will start exposing it and writing documentation once it is declared stable enough.

https://github.com/Minestom/Minestom/tree/master/src/test/java/net/minestom/server/api
Notice the test files ending with `IntegrationTest` for usage example.

**Comment by BomBardyGamer** [comment]
You could try using [canary](https://github.com/mworzala/canary), though I'm not sure if it's updated. @mworzala can probably provide more information.

**Reply by mworzala** [reply]
Not tested on the most recent version (but probably fine). Will likely just be merged into the internal one when I have more time.

**Comment by ItsLiyua** [comment]
Is there a testing API by now or is it still internal?
I'd like to write some tests.

**Reply by TheMode** [reply]
You can use our `testing` module https://github.com/Minestom/Minestom/tree/master/testing.

There is no wiki page yet (contribution is possible) but you can take a look at all our files ending with `*IntegrationTest`
